src/core/components/AuthenticateButton/index.js

```diff
@@ -53,9 +55,17 @@ export const mapStateToProps = (state) => ({
 });
 
 export const mapDispatchToProps = (dispatch) => ({
-  handleLogOut() {
+  handleLogOut({ api }) {
+    const { location } = window;
+    if (config.get('apiHost') === `${location.protocol}//${location.host}`) {
+      // Deployed config, server manages cookie.
+      return logOutFromServer({ api })
+        .then(() => dispatch(logOutUser()));
```

Comment:
dang, we really need to see global errors. Looks like a failure here will just die in the console. https://github.com/mozilla/addons-frontend/issues/1484